src/Features/CSharp/Portable/CodeFixes/PopulateSwitch/PopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PopulateSwitch
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddLabelsToSwitch), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ImplementInterface)]
+    internal partial class PopulateSwitchCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(IDEDiagnosticIds.PopulateSwitchDiagnosticId); }
+        }
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return PopulateSwitchFixAllProvider.Instance;
+        }
+
+        private static SwitchStatementSyntax GetSwitchStatementNode(SyntaxNode root, TextSpan span)
+        {
+            var token = root.FindToken(span.Start);
+            if (!token.Span.IntersectsWith(span))
```

Comment:
I have absolutely no idea.  I took the general structure for this class from the RemoveUnnecessaryCast analyzer base.